first-party/jib-classpath-main-class-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/classpathmainclass/ClasspathMainClassExtractor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle.extension.classpathmainclass;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public class ClasspathMainClassExtractor {
+
+  private static List<String> classpathOptions = Arrays.asList("-cp", "-classpath", "--class-path");
+
+  static Map<String, String> extractEnvsFromEntrypoint(@Nullable List<String> entrypoint) {
+    if (entrypoint == null || entrypoint.size() == 0) {
+      return Collections.emptyMap();
+    }
+
+    if (!entrypoint.get(0).equals("java")) {
+      return Collections.emptyMap();
+    }
+
+    int i = entrypoint.size() - 1;
+    final String mainClass = entrypoint.get(i);
```

Comment:
nit: we don't use `final` for local variables unless absolutely necessary.
```suggestion
    String mainClass = entrypoint.get(i);
```

Reply:
findbugs (?) forced me to do this...

Reply:
Alright, no worries. I'll take care of this later.